language/functional_tests/src/evaluator.rs

```diff
@@ -315,6 +323,9 @@ pub fn eval(config: &GlobalConfig, transactions: &[Transaction]) -> Result<Evalu
         exec.add_account_data(&data);
     }
 
+    // Now add in the genesis info
+    add_genesis_account_info(&mut config, &exec);
```

Comment:
`fn add_genesis_account_info` is doing a lot of unnecessary work. It's also unfortunate you need to take the ownership of `config` and inject the genesis account data into it, but I can see the rationale.

I think this is a better way:
1) Change the type of `genesis_accounts ` in `global::Config` from `BTreeMap<String, AccountAddress>` to `BTreeMap<String, Account>`. To get the association account, use `e2e_tests::account::new_association`.
2) To get the address of an account, look up both `config.accounts` & `config.genesis_accounts`. Might be worth to make a helper function for this.
3) Change `fn make_script_transaction` & `fn make_module_transaction` to take an `Account` instead of `AccountData`. It only uses the `Account` part anyway.

This way you avoid reading the `AccountData` for the association from the fake executor and can keep the config immutable.